File: dataset/ourMethod/CLAUDE_claude-sonnet-4-5-20250929/Task22.java

import java.util.ArrayList;
import java.util.List;

public class Task22 {
    public static List<Integer> filterList(List<Object> list) {
        if (list == null) {
            throw new IllegalArgumentException("Input list cannot be null");
        }
        
        List<Integer> result = new ArrayList<>();
        
        for (Object element : list) {
            if (element == null) {
                continue;
            }
            
            if (element instanceof Integer) {
                int value = (Integer) element;
                if (value >= 0) {
                    result.add(value);
                }
            } else if (element instanceof Long || element instanceof Short || element instanceof Byte) {
                long value = ((Number) element).longValue();
                if (value >= 0 && value <= Integer.MAX_VALUE) {
                    result.add((int) value);
                }
            }
        }
        
        return result;
    }
    
    public static void main(String[] args) {
        System.out.println("Test 1: " + filterList(List.of(1, 2, "a", "b")));
        System.out.println("Test 2: " + filterList(List.of(1, "a", "b", 0, 15)));
        System.out.println("Test 3: " + filterList(List.of(1, 2, "a", "b", "aasf", "1", "123", 123)));
        
        List<Object> test4 = new ArrayList<>();
        test4.add(5);
        test4.add("test");
        test4.add(10);
        test4.add(null);
        test4.add(20);
        System.out.println("Test 4: " + filterList(test4));
        
        System.out.println("Test 5: " + filterList(List.of(100, 200, "xyz", 300)));
    }
}

File: dataset/ourMethod/CLAUDE_claude-sonnet-4-5-20250929/Task22.py

from typing import List, Any, Union


def filter_list(lst: List[Any]) -> List[int]:
    if lst is None:
        raise ValueError("Input list cannot be None")
    
    if not isinstance(lst, list):
        raise TypeError("Input must be a list")
    
    result = []
    
    for element in lst:
        if element is None:
            continue
        
        if isinstance(element, bool):
            continue
        
        if isinstance(element, int):
            if element >= 0:
                result.append(element)
    
    return result


def main():
    print("Test 1:", filter_list([1, 2, "a", "b"]))
    print("Test 2:", filter_list([1, "a", "b", 0, 15]))
    print("Test 3:", filter_list([1, 2, "a", "b", "aasf", "1", "123", 123]))
    print("Test 4:", filter_list([5, "test", 10, None, 20]))
    print("Test 5:", filter_list([100, 200, "xyz", 300]))


if __name__ == "__main__":
    main()
